controller/admin_transactions_controller.php

```diff
@@ -395,6 +495,113 @@ public function get_valid_offset()
 		return ($this->last_page_offset) ? (int) $this->last_page_offset : 0;
 	}
 
+	/**
+	 * Prepare data array() before send it to $entity
+	 *
+	 * @param array $transaction_data
+	 *
+	 * @return array
+	 * @throws transaction_exception
+	 */
+	private function build_data_ary($transaction_data)
+	{
+		$errors = array();
+
+		if ($this->request->is_set('u') && $transaction_data['MT_USERNAME'] === '')
+		{
+			$user_id = ANONYMOUS;
+		}
+		else
+		{
+			$user_ary = $this->ppde_operator->query_donor_user_data('username', $transaction_data['MT_USERNAME']);
+
+			if ($user_ary)
+			{
+				$user_id = $user_ary['user_id'];
+			}
+			else
+			{
+				$errors[] = $this->language->lang('PPDE_MT_DONOR_NOT_FOUND', $transaction_data['MT_USERNAME']);
+			}
+		}
+
+		if ($transaction_data['MT_MC_GROSS'] <= 0)
+		{
+			$errors[] = $this->language->lang('PPDE_MT_MC_GROSS_TOO_LOW');
+		}
+
+		if ($transaction_data['MT_MC_FEE'] < 0)
+		{
+			$errors[] = $this->language->lang('PPDE_MT_MC_FEE_NEGATIVE');
+		}
+
+		if ($transaction_data['MT_MC_FEE'] >= $transaction_data['MT_MC_GROSS'])
+		{
+			$errors[] = $this->language->lang('PPDE_MT_MC_FEE_TOO_HIGH');
+		}
+
+		$payment_date = implode('-', [
+			$transaction_data['MT_PAYMENT_DATE_YEAR'],
+			$transaction_data['MT_PAYMENT_DATE_MONTH'],
+			$transaction_data['MT_PAYMENT_DATE_DAY'],
+		]);
+
+		$payment_date_timestamp_at_midnight = $this->user->get_timestamp_from_format('Y-m-d H:i:s', $payment_date . ' 00:00:00');
+
+		if ($payment_date_timestamp_at_midnight === false)
```

Comment:
How can we enter in its condition?
A tried do enter invalid `$payment_date`, such as `2018-13-13` and it always return a timestamp.
`2018-13-13` is remplaced by `2019-01-13`

Do we disallow donation newer than the current timestamp? This will make sense as we cannot receive a donation a date not yet due. (not sure for this sentence, but I hope you will understand what I mean 😃).

Reply:
Yes I will add a check for that.

The date picker doesn't allow picking a date in the future, so just to be clear, you entered the future date in the 3 date fields, not in the date picker, correct? If you used the date picker you found another bug.

Reply:
> Yes I will add a check for that.
> 
> The date picker doesn't allow picking a date in the future, so just to be clear, you entered the future date in the 3 date fields, not in the date picker, correct? If you used the date picker you found another bug.

Yes, this is not throught the date picker.

> I don't know when exactly `DateTime::createFromFormat` returns `false` but for example it does when you use a 3 digits month or day.
I haven't tested this case, but it make sense.